extensions/git/src/git.ts

```diff
@@ -697,6 +700,38 @@ export class Repository {
 		});
 	}
 
+	async getLog(options?: GitLogOptions): Promise<Commit[]> {
+		const args = ['log'];
+		if (options) {
+			if (typeof options.maxEntries === 'number' && options.maxEntries > 0) {
+				args.push('-' + options.maxEntries);
+			}
+		}
+
+		args.push(`--pretty=format:${COMMIT_FORMAT}%x00%x00`);
+		const gitResult = await this.run(args);
+		if (gitResult.exitCode) {
+			// An empty repo.
+			return [];
+		}
+
+		const entries = gitResult.stdout.split('\x00\x00');
```

Comment:
This will get expensive. We should instead do a _while regex exec_ loop and keep matching commits until we reach the end of `gitResult.stdout`.